courses-rest/src/main/java/no/sikt/nva/fs/CoursesByInstitutionOfLoggedInUserHandler.java

```diff
@@ -14,34 +13,39 @@
 import no.unit.nva.commons.json.JsonUtils;
 import nva.commons.apigateway.ApiGatewayHandler;
 import nva.commons.apigateway.RequestInfo;
+import nva.commons.apigateway.exceptions.ApiGatewayException;
 import nva.commons.core.Environment;
 import nva.commons.core.JacocoGenerated;
-import nva.commons.core.attempt.Try;
 import nva.commons.core.paths.UriWrapper;
+import nva.commons.secrets.SecretsReader;
 
 public class CoursesByInstitutionOfLoggedInUserHandler extends ApiGatewayHandler<Void, CoursesResponse> {
 
-    public static final ObjectMapper OBJECT_MAPPER = JsonUtils.dtoObjectMapper;
-    /* default */ static final String FS_CONFIG_ENV_NAME = "FS_CONFIG";
+    /* default */ static final String FS_CONFIG_SECRET_NAME_ENV_NAME = "FsConfigSecretName";
+    /* default */ static final String FS_CONFIG_NOT_PROPERLY_FORMATTED_JSON = "FS configuration from secrets manager "
+                                                                              + "does not contain properly formatted "
+                                                                              + "JSON!";
 
-    private final FsConfig fsConfig;
     private final TimeProvider timeProvider;
+    private final SecretsReader secretsReader;
 
     @JacocoGenerated
     public CoursesByInstitutionOfLoggedInUserHandler() {
-        this(new Environment(), Clock.system(ZoneId.systemDefault()));
+        this(new Environment(), new SecretsReader(), Clock.system(ZoneId.systemDefault()));
     }
 
-    public CoursesByInstitutionOfLoggedInUserHandler(final Environment environment, final Clock clock) {
+    public CoursesByInstitutionOfLoggedInUserHandler(final Environment environment,
+                                                     final SecretsReader secretsReader,
```

Comment:
Better to have the dependency on the AWS SecretClient instead on the custom nva-commons resource. The nva commons may disappear at some point an it would be better to not have the API depending on such a class.
For the default constructor there is a method SecretsReader.defaultSecretsClient

Reply:
I will fix.